devito/function.py

```diff
@@ -669,19 +653,27 @@ def laplace(self):
         derivative wrt. all spatial dimensions.
         """
         derivs = tuple('d%s2' % d.name for d in self.space_dimensions)
-
         return sum([getattr(self, d) for d in derivs[:self.ndim]])
 
     def laplace2(self, weight=1):
         """
         Generates a symbolic expression for the double Laplacian
         wrt. all spatial dimensions.
         """
-        order = self.space_order/2
-        first = sum([second_derivative(self, dim=d, order=order)
-                     for d in self.space_dimensions])
-        return sum([second_derivative(first * weight, dim=d, order=order)
-                    for d in self.space_dimensions])
+        derivs = tuple('d%s2' % d.name for d in self.space_dimensions)
+        return sum([getattr(self.laplace * weight, d) for d in derivs[:self.ndim]])
+
+    def getdiff(self, name):
```

Comment:
maybe you actually want to override ``__getattr__``?
see for example here: https://github.com/opesci/devito/blob/master/devito/yask/data.py#L216

Reply:
I tried, but some `sympy.Expr` properties are needed, and I ended up with circular infite calls (why the unused `get_diff`)